src/web/dosguard/DOSGuard.hpp

```diff
@@ -59,6 +61,7 @@ class DOSGuard : public DOSGuardInterface {
     util::Mutex<State> mtx_;
 
     std::reference_wrapper<WhitelistHandlerInterface const> whitelistHandler_;
+    std::reference_wrapper<WeightsInterface const> weights_;
```

Comment:
Who is responsible for storage of `weights_`?
I think with this implementation it's easy to have a reference to a local object which goes out of scope.

Maybe we could store `unique_ptr` to the interface instead?

Reply:
It will be not so easy to provide a mock with `unique_ptr`. 

Dosguard has such design for all its components, I created an issue to fix all of them: #2083.

Reply:
Thanks!